homeassistant/components/purpleair/config_flow.py

```
@@ -71,6 +93,77 @@ def async_get_nearby_sensors_schema(options: list[SelectOptionDict]) -> vol.Sche
     )
 
 
+@callback
+def async_get_remove_sensor_options(
+    hass: HomeAssistant, config_entry: ConfigEntry
+) -> list[SelectOptionDict]:
+    """Return a set of already-configured sensors as SelectOptionDict objects."""
+    device_registry = dr.async_get(hass)
+    return [
+        SelectOptionDict(value=device_entry.id, label=cast(str, device_entry.name))
+        for device_entry in device_registry.devices.values()
+        if config_entry.entry_id in device_entry.config_entries
+    ]
+
+
+@callback
+def async_get_remove_sensor_schema(sensors: list[SelectOptionDict]) -> vol.Schema:
+    """Define a schema removing a sensor."""
+    return vol.Schema(
+        {
+            vol.Required(CONF_SENSOR_DEVICE_ID): SelectSelector(
+                SelectSelectorConfig(options=sensors, mode=SelectSelectorMode.DROPDOWN)
+            )
+        }
+    )
+
+
+@callback
+def async_get_sensor_index(
+    hass: HomeAssistant, config_entry: ConfigEntry, device_entry: dr.DeviceEntry
+) -> int:
+    """Get the sensor index related to a config and device entry.
+
+    Note that this method expects that there will always be a single sensor index per
+    DeviceEntry.
+    """
+    device_registry = dr.async_get(hass)
+    device_entry = cast(dr.DeviceEntry, device_registry.async_get(device_entry.id))
+
+    [sensor_index] = [
+        sensor_index
+        for sensor_index in config_entry.options[CONF_SENSOR_INDICES]
+        if (DOMAIN, str(sensor_index)) in device_entry.identifiers
+    ]
+
+    return cast(int, sensor_index)
+
+
+@callback
+def async_remove_sensor_by_device_id(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    device_id: str,
+    *,
+    remove_device: bool = True,
+) -> dict[str, Any]:
+    """Remove a sensor and return update config entry options."""
+    device_registry = dr.async_get(hass)
+    device_entry = cast(dr.DeviceEntry, device_registry.async_get(device_id))
```

Should we raise here if the entry is missing?